### PR TITLE
feat(connext): reject app install for transfers without status field

### LINKED ISSUE #1850: Call `/reject-install` for payments without status
We confirmed that payments without status are merely install proposals (that failed for some reason) and that need to be cleaned up. So what should happen if `xud` gets an empty status field for a payment hash like this: it should call `/reject-install` for this appIdentityHash. Then run `/hashlock-resolve` which should give the expected `404`.

Edit by karl: remove paymentId -> appIdentityHash

### PR DESCRIPTION
**Pull request by ghost**
This PR changes the logic around how `Connext.lookupPayment` handles transfer statuses without a status field.

Previously, we'd always consider the payment as pending. We're now attempting to reject the app proposal install. If successful, we're considering the payment as permanently failed.

Closes https://github.com/ExchangeUnion/xud/issues/1850

### CODE REVIEW

**Comment by raladev** [comment]
waiting for https://github.com/connext/rest-api-client/issues/92

**Comment by kilrau** [comment]
> waiting for [connext/rest-api-client#92](https://github.com/connext/rest-api-client/issues/92)

Merged, can be tested against https://github.com/connext/rest-api-client/releases/tag/1.3.3

File: lib/connextclient/ConnextClient.ts [hunk 1]
```diff
@@ -470,8 +470,17 @@ class ConnextClient extends SwapClient {
...
           return { state: PaymentState.Failed };
         default:
-          this.logger.debug(`no hashlock status for payment with hash ${rHash}: ${JSON.stringify(transferStatusResponse)}`);
-          return { state: PaymentState.Pending };
+          this.logger.debug(`no hashlock status for payment with hash ${rHash}: ${JSON.stringify(transferStatusResponse)} - attempting to reject app install`);
+          try {
+            await this.sendRequest('/reject-install', 'POST', {
+              appIdentityHash: transferStatusResponse.appIdentityHash,
+            });
+            return { state: PaymentState.Failed };
```
Comment (sangaman) [inline lib/connextclient/ConnextClient.ts:478]: Maybe log a line here right before returning indicating that we rejected the install and the payment status is failed?

**Review by sangaman — DISMISSED** [review]
Awesome, this is an important fix.

File: lib/connextclient/ConnextClient.ts (hunk 1 above)
Comment (erkarl) [reply lib/connextclient/ConnextClient.ts:479]: I've added a `debug` level log here and cleaned up the terminology in this function to be in line with connext.

**Comment by erkarl** [comment]
This PR requires us to update the connext client to 1.3.3 and node to latest.

**Review by sangaman — DISMISSED** [review]
LGTM!

**Comment by raladev** [comment]
- [ ] looks like something wrong with request building
```
07/09/2020 12:58:47.365 [CONNEXT] trace: hashlock status for connext transfer with hash 78fd7f8c20b16263196a01c3b0c6e7b6f80b14f430339db2882d2c2aa7e4e6b0 is undefined
07/09/2020 12:58:47.366 [CONNEXT] debug: no hashlock status for connext transfer with hash 78fd7f8c20b16263196a01c3b0c6e7b6f80b14f430339db2882d2c2aa7e4e6b0: {"senderIdentifier":"indra7TdM96xqgvso2oG3zNyaoN7xebyfHMskPb5Fp8bffk8R8qYqkk","receiverIdentifier":"indra7GFej81aBFjQjEeStYCKbf8mqWD5N39k4Y8QSfwzh4zjcArhP5","senderAppIdentityHash":"0x66f3ea4db257993d7e5905a4c86cf69a172cdb98f55a8f75579becb8b3168aee","assetId":"0x0000000000000000000000000000000000000000","amount":"1000000000000000000","lockHash":"0x78fd7f8c20b16263196a01c3b0c6e7b6f80b14f430339db2882d2c2aa7e4e6b0","expiry":"154671"} - attempting to reject app install proposal
07/09/2020 12:58:47.367 [CONNEXT] trace: sending request to /reject-install: {}
07/09/2020 12:58:47.376 [CONNEXT] error: connext server error 500: Internal Server Error
07/09/2020 12:58:47.384 [CONNEXT] error: failed to reject connext app install proposal: 8.5 - connext server error 500: Internal Server Error
```

```
[1599483527637] INFO  (70 on connext): Getting client for publicIdentifier: indra7TdM96xqgvso2oG3zNyaoN7xebyfHMskPb5Fp8bffk8R8qYqkk
2020-09-07T12:58:47.724Z [NodeApiClient] Node responded to indra7TdM96xqgvso2oG3zNyaoN7xebyfHMskPb5Fp8bffk8R8qYqkk.indra6WRA6HQQTDQD1FC3KmdbpePijdcgLTqAdi2xy2Q2HqNeGn4cBh.1337.transfer.get-hashlock request in 87 ms
[1599483527724] INFO  (70 on connext): request completed
    url: "/hashlock-status/0x7a8a75ef8925e97f6e05f5394868148c053ef7e878feaa67d2dd7821d7331068/0x0000000000000000000000000000000000000000"
    statusCode: 200
    reqId: 860
[1599483527726] INFO  (70 on connext): received request
    url: "/reject-install"
    id: 861
    reqId: 861
[1599483527726] ERROR (70 on connext): Invalid or missing appIdentityHash
    Error: Invalid or missing appIdentityHash
        at Object.requireParam (/app/build/src/helpers/utilities.js:51:15)
        at /app/build/src/index.js:781:31
        at step (/app/build/src/index.js:44:23)
        at Object.next (/app/build/src/index.js:25:53)
        at /app/build/src/index.js:19:71
        at new Promise (<anonymous>)
        at __awaiter (/app/build/src/index.js:15:12)
        at Object.<anonymous> (/app/build/src/index.js:775:106)
        at preHandlerCallback (/app/node_modules/fastify/lib/handleRequest.js:120:28)
        at Auth.next [as done] (/app/node_modules/fastify/lib/hooks.js:158:7)
[1599483527727] INFO  (70 on connext): request completed
    url: "/reject-install"
    statusCode: 500
    reqId: 861
[1599483528209] INFO 
```

**Review by raladev — CHANGES_REQUESTED** [review]
above

**Review by raladev — APPROVED** [review]

**Pull request merged**